### PR TITLE
Eli/fixed price v2

### PR DESCRIPTION
**Pull request by EGuenz**
1. Restructuring of fixed-price storage 
2. Switch to nat keyed big_map
3. Checks empty transfers (e.g. sale-price of 0) so that operation does not fail
4. Adds amount field to allow for sale of an FT based edition set where each FT is sold at the same price
5. Standardizes checks of fee being too high across contracts.
6. Updates Taquito/Cleveland tests
7. Updates Docs
8. Remove money token check from fa2 fixed price allowlist
9. Added getActiveSales views for tez and fa2 sale contracts
10. Changes fee calculation to stop unnecessarily using Ceiling to avoid edge cases in calculation

### CODE REVIEW

**Review by emishur — APPROVED** [review]

File: packages/minter-contracts/ligo/src/fixed_price_sale/README.md [hunk 1]
```diff
@@ -217,3 +226,19 @@ Respective allowlisted contract is [`fixed_price_sale_market_tez_allowlisted.mli
...
+``` ocaml
+type fee_data = 
+  [@layout:comb]
+  {
+    fee_address : address;
+    fee_percent : nat;
+  }
+```
+
+`fee_percent` is a nat between 0 and 100 representing the percent of the sale to be paid to the `fee_address`. If `fee_percent > 100`, the `sell` and `buy` entrypoints will both fail with `FEE_TOO_HIGH`.  Fee is calculated as `((fee_percent * price) / 100)`.
```
Comment (Martoon-00) [inline packages/minter-contracts/ligo/src/fixed_price_sale/README.md:244]: Great that this is mentioned in the spec!

One thing I would also explicitly mention: there are two possible models of how fees are accounted for. Let's say fees form 30%, and the seller declared a price of 100 money; whether will `(fees, seller_received, buyer_payed)` be `(30, 100, 130)` (fees are added to the declared price) or `(30, 70, 100)` (fees are already included into the price) here?

I see the answers to this question in the code but would be really handy to have it mentioned in the specification too.

File: packages/minter-contracts/ligo/src/fixed_price_sale/fixed_price_sale_market_allowlisted.mligo [hunk 2]
```diff
@@ -14,7 +14,6 @@ let check_allowed_param (param, allowlist : market_entry_points * allowlist) : u
   match param with
   | Sell p -> begin
       check_address_allowed (p.sale_token.fa2_address, allowlist, "SALE_ADDRESS_NOT_ALLOWED");
-      check_address_allowed (p.money_token.fa2_address, allowlist, "MONEY_ADDRESS_NOT_ALLOWED");
```
Comment (Martoon-00) [inline packages/minter-contracts/ligo/src/fixed_price_sale/fixed_price_sale_market_allowlisted.mligo]: If you are removing this, could you also update the `README` on this, and remove the `moneyAddressNotAllowed` constant from Haskell tests and the code that checks for this error?

File: packages/minter-contracts/src-hs/Lorentz/Contracts/Marketplace/Tez.hs [hunk 3]
```diff
@@ -12,49 +12,58 @@ import qualified Michelson.Typed as T
...
-initMarketplaceTezStorage as = MarketplaceTezStorage as mempty
+initMarketplaceTezStorage as = MarketplaceTezStorage as mempty (SaleId 0) 
 
 data MarketplaceTezEntrypoints
-  = Sell InitSaleParamTez
-  | Buy SaleParamTez
-  | Cancel SaleParamTez
+  = Sell SaleDataTez
+  | Buy SaleId
+  | Cancel SaleId
```
Comment (Martoon-00) [inline packages/minter-contracts/src-hs/Lorentz/Contracts/Marketplace/Tez.hs:66]: :+1:

**Comment by EGuenz** [comment]
Closes #127

**Pull request merged**